redash/app.py

```diff
@@ -0,0 +1,62 @@
+from flask import Flask
+from flask_sslify import SSLify
+from werkzeug.contrib.fixers import ProxyFix
+
+from . import mail, settings
+from .utils import routes
+
+
+class Redash(Flask):
+    """A custom Flask app for Redash"""
+    def __init__(self, *args, **kwargs):
+        kwargs.update({
+            'template_folder': settings.STATIC_ASSETS_PATH,
+            'static_folder': settings.STATIC_ASSETS_PATH,
+            'static_path': '/static',
+        })
+        super(Redash, self).__init__(__name__, *args, **kwargs)
+        # Make sure we get the right referral address even behind proxies like nginx.
+        self.wsgi_app = ProxyFix(self.wsgi_app, settings.PROXIES_COUNT)
+        # Handle the special case of the org slug
+        self.url_map.converters['org_slug'] = routes.SlugConverter
+        # Configure Redash using our settings
+        self.config.from_object('redash.settings')
+
+
+def create_app():
+    from . import authentication, extensions, handlers, limiter, migrate
+    from .destinations import import_destinations
+    from .handlers import chrome_logger
+    from .handlers.webpack import configure_webpack
+    from .metrics.request import provision_app
+    from .models import db, users
+    from .query_runner import import_query_runners
+    from .utils import sentry
+    from .version_check import reset_new_version_status
+
+    sentry.init()
+    app = Redash()
+
+    # Check and update the cached version for use by the client
+    app.before_first_request(reset_new_version_status)
+
+    # Load query runners and destinations
+    import_query_runners(settings.QUERY_RUNNERS)
+    import_destinations(settings.DESTINATIONS)
```

Comment:
This moves importing the query runners and destinations into a bit later during startup, and ties them to the app creation process. I think this is more in line what we expect since both can trigger all kinds of imports in the various backend code.